hierarchicalforecast/methods.py

```diff
@@ -11,10 +11,9 @@
 
 import numpy as np
 from numba import njit
-
+from quadprog import solve_qp
```

Comment:
That's the first thing I tried. The problem is that we are dealing with weighted NNLS instead of pure NNLS. The weights are provided by the inverse of the variance-covariance matrix. As stated in the documentation, `scipy.optimize.nnls` solves the problem,

```
min x || Ax - b|| st x>=0
```

If we want to include weights, we have to solve the problem,

```
min x || W^(-1/2)Ax - W^(-1/2)b|| st x>=0
```

Observe that, in this case, we need to calculate the squared root of `W^(-1/2)`. When `W` is diagonal, for example, `ols` or `wls_struct`, there is no problem. But with other methods, such as `mint_shrink`, it's impossible to guarantee the positiveness of each entry in `W`.  

That's the reason behind the election of `quadprog`.

Reply:
Would it be possible to use Cholesky matrix and premultiply NNL inputs?

Reply:
Positive definite W =/> `W_{ij} > 0` for all i,j

Reply:
I am getting confused, are we actually computing the sqrt of the entries of the matrix?
I thought the algorithm was using the sqrt of the inverse matrix, ie. Cholesky's matrix
https://en.wikipedia.org/wiki/Square_root_of_a_matrix

Reply:
hey @kdgutier. I've updated the code to include comments about the decision behind `quadprog`,

https://github.com/Nixtla/hierarchicalforecast/blob/a5c1a7968442984477aaaf764a592a840619d9ee/hierarchicalforecast/methods.py#L474-L489